File: aoc2022/day1/day1-1.m
#import <Foundation/Foundation.h>

int main() {
    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
    // NSString* filePath = @"input.txt";
    NSString* fileContent = [NSString stringWithContentsOfFile:@"../input.txt"
                            encoding:NSUTF8StringEncoding error:nil];
    NSArray* lines = [fileContent componentsSeparatedByString:(NSString *)@"\n"];
    int currentHigh = 0;
    int runningTotal = 0;
    for (NSString* line in lines) {
        // NSLog(@"line: %@", line);
        if ([line length] > 0) {
            int currentNum = [line intValue];
            runningTotal = runningTotal + currentNum; 
            NSLog(@"currentNum: %d", currentNum); 
            NSLog(@"runningTotal: %d", runningTotal); 
        } else {
            if (runningTotal > currentHigh) {
                currentHigh = runningTotal;
                NSLog(@"currentHigh: %d", currentHigh);
            }
            runningTotal = 0;
        }
    }
    NSLog(@"currentHigh: %d", currentHigh);

    [pool drain];
    return 0;
}

File: aoc2022/day1/day1-1.swift
#!/usr/bin/env swift
import Foundation

let filePath = "input.txt"
let fullPath = NSString(string: filePath).expandingTildeInPath
 do
 {
    let fileContent = try NSString(contentsOfFile: filePath, encoding: String.Encoding.utf8.rawValue)
    print(fileContent)
    let lines = fileContent.components(separatedBy:"\n")
    var runningTotal = 0
    var highestTotal = 0
    for line in lines {
        print(line)
        if line.count < 1 {
            if runningTotal > highestTotal {
                highestTotal = runningTotal
            }
            runningTotal = 0
        } else {
            let currentVal = Int(line)
            runningTotal = runningTotal + currentVal!
        }
    }
     print("highest total: \(highestTotal)")
 }
 catch
 {
     print(error)
 }